lib/services/local/generic/generic_test.go

```diff
@@ -39,6 +43,7 @@ import (
 // testResource for testing the generic service.
 type testResource struct {
 	types.ResourceHeader
+	SpecPropA string
```

Comment:
Use struct type as the spec and call the field Spec, so it's clearer what this means?